pkg/sink/sink.go

```diff
@@ -147,45 +148,17 @@ func (r Sink) HandleEvent(response http.ResponseWriter, request *http.Request) {
 		response.WriteHeader(http.StatusInternalServerError)
 		return
 	}
-	result := make(chan int, 10)
 	// Execute each Trigger
+	r.WaitGroup.Add(len(triggers))
 	for _, t := range triggers {
 		go func(t triggersv1.Trigger) {
+			defer r.WaitGroup.Done()
 			localRequest := request.Clone(request.Context())
-			if err := r.processTrigger(t, localRequest, event, eventID, eventLog); err != nil {
```

Comment:
So yeah it still fails.  The user presumably learns of it from some other means.  Full disclosure, I did not research *how* that might now be or not be achieved.

If there is an easy enough reader's digest explanation @jmcshane @dibyom please elaborate.  Or if a pointer to what I have to read to understand makes more sense, that works too.

thanks

Reply:
I think the diff for the docs/eventlisteners.md page shows the change: https://github.com/tektoncd/triggers/pull/1077/files#diff-7da54b4f5484ec784351290e106d6aebea551a17046b6116cc8e65964ebb693f

i.e instead of waiting for a resource to be created and then responding with a 201 status code, Triggers will now return a 202 response earlier (i.e. when it has selected the triggers to process -- it will no longer wait for the trigger processing to complete)

Reply:
OK so yeah that markdown, assuming basic HTTP rc knowledge, makes it clear that the processing here is now asynchronous vs. synchronous.

With all the scalability advantages, etc. obtained with that shift.

In theory then the poster of the event then needs to know what was triggered by the event and then query those artifacts directly for status.

Quick code scan / memory refresh and I see the `/triggers-eventid` label set on the objects created (which we have had for a while) facilitates that query.

Would a cross reference to https://github.com/tektoncd/triggers/blob/4b9aa3157f3989255f350c8c60c2b6aecab38783/docs/triggertemplates.md#structure-of-a-triggertemplate somewhere and its explanation of the label near https://github.com/tektoncd/triggers/pull/1077/files#diff-7da54b4f5484ec784351290e106d6aebea551a17046b6116cc8e65964ebb693f be a useful change in your opinion @dibyom ?

Other than that potential tweak, I'm good.  Thanks.

Reply:
Ok yeah, I think adding a couple of lines or (maybe link to a different section) with the information on querying makes sense.